tests/test_mbed_targets.py

```diff
@@ -256,58 +202,40 @@ def test_get_target_failure(self, mocked_get_target_data):
         ]
-        mocked_get_target_data.return_value = fake_target_data
-
-        mbed_targets = MbedTargets.from_offline_database()
-        json_str = mbed_targets.json_dump()
-
-        self.assertEqual(
-            json.loads(json_str), fake_target_data, "Deserialised JSON string should match original target data"
-        )
-
-    @mock.patch("mbed_targets._internal.target_database.get_online_target_data")
-    def test_subtracting_targets_takes_difference_non_symmetric(
-        self, mocked_online_target_data, mocked_offline_target_data
-    ):
-        tgt_data_off = _make_dummy_internal_target_data()
-        tgt_data_on = _make_dummy_internal_target_data()
-        tgt_data_off.pop()
-        mocked_offline_target_data.return_value = tgt_data_off
-        mocked_online_target_data.return_value = tgt_data_on
-
-        tgts_off = MbedTargets.from_offline_database()
-        tgts_on = MbedTargets.from_online_database()
-        diff_base_on = tgts_on - tgts_off
-        diff_base_off = tgts_off - tgts_on
+        mocked_get_online_target_data.return_value = raw_target_data
 
-        self.assertEqual(len(diff_base_on), 1)
-        self.assertEqual(len(diff_base_off), 0)
+        targets = [MbedTarget.from_target_entry(t) for t in raw_target_data]
+        filtered_target_data = [target.__dict__ for target in targets]
+        mocked_get_offline_target_data.return_value = filtered_target_data
 
-    def test_subtracting_targets_returns_targets_instance(self, mocked_offline_target_data):
-        mocked_offline_target_data.return_value = _make_dummy_internal_target_data()
+        # MbedTargets.from_online_database handles "raw" target entries from the online db
+        mbed_targets = MbedTargets.from_online_database()
+        json_str_from_raw = mbed_targets.json_dump()
+        t1_raw, t2_raw = mbed_targets
 
-        tgts_one = MbedTargets.from_offline_database()
-        tgts_two = MbedTargets.from_offline_database()
+        # MbedTargets.from_offline_database expects the data to have been "filtered" through the MbedTargets interface
+        mbed_targets_offline = MbedTargets.from_offline_database()
+        json_str_from_filtered = mbed_targets_offline.json_dump()
+        t1_filt, t2_filt = mbed_targets_offline
 
-        self.assertTrue(isinstance(tgts_one - tgts_two, MbedTargets))
-
-    def test_difference_operator_raises_for_non_targets_instance(self, mocked_offline_target_data):
-        mocked_offline_target_data.return_value = _make_dummy_internal_target_data()
-
-        tgts = MbedTargets.from_offline_database()
+        self.assertEqual(
+            json_str_from_raw,
+            json.dumps([t1_raw.__dict__, t2_raw.__dict__], indent=4),
+            "JSON string should match serialised target __dict__.",
+        )
 
-        with self.assertRaises(TypeError):
-            tgts - 1
+        self.assertEqual(json_str_from_filtered, json.dumps([t1_filt.__dict__, t2_filt.__dict__], indent=4))
```

Comment:
Is `__dict__` equivalent to `asdict`? If so, I'd rather we avoid using dunders where possible.